### PR TITLE
Change unsubjet_map in btag info producers to tracked

### PR DESCRIPTION
**Pull request by stahlleiton**
#### PR description:

This PR change the unsubjet_map variable in the BTag info producers to tracked as requested in https://github.com/cms-sw/cmssw/pull/44422#discussion_r1548089749

### CODE REVIEW

**Comment by mandrenguyen** [comment]
please test

**Comment by mandrenguyen** [comment]
type btv

**Comment by mandrenguyen** [comment]
+1

**Comment by mmusich** [comment]
@stahlleiton please address https://github.com/cms-sw/cmssw/pull/44422#discussion_r1548230840.
Using this PR on top of our (TSG) development branches in which the HLT template was already migrated to CMSSW_14_0_4 (containing the erroneous https://github.com/cms-sw/cmssw/pull/44422) leads to failed validation of the configuration:

```
----- Begin Fatal Exception 03-Apr-2024 08:54:22 CEST-----------------------
An exception of category 'Configuration' occurred while
   [0] Constructing the EventProcessor
   [1] Validating configuration of module: class=DeepBoostedJetTagInfoProducer label='hltParticleNetJetTagInfos'
Exception Message:
In the configuration, parameter "unsubjet_map" is defined as an untracked InputTag.
It should be tracked.
----- End Fatal Exception -------------------------------------------------
``` 

please do as suggested by @Martin-Grunewald:

>  Thus also an entry in the customisation routine is needed to just remove the parameter by name.

**Comment by stahlleiton** [comment]
> @stahlleiton please address [#44422 (comment)](https://github.com/cms-sw/cmssw/pull/44422#discussion_r1548230840). Using this PR on top of our (TSG) development branches in which the HLT template was already migrated to CMSSW_14_0_4 (containing the erroneous #44422) leads to failed validation of the configuration:
> 
> ```
> ----- Begin Fatal Exception 03-Apr-2024 08:54:22 CEST-----------------------
> An exception of category 'Configuration' occurred while
>    [0] Constructing the EventProcessor
>    [1] Validating configuration of module: class=DeepBoostedJetTagInfoProducer label='hltParticleNetJetTagInfos'
> Exception Message:
> In the configuration, parameter "unsubjet_map" is defined as an untracked InputTag.
> It should be tracked.
> ----- End Fatal Exception -------------------------------------------------
> ```
> 
> please do as suggested by @Martin-Grunewald:
> 
> > Thus also an entry in the customisation routine is needed to just remove the parameter by name.

I tried to address Martin's comment. Let me know if it is fine or if I should change something in: https://github.com/cms-sw/cmssw/pull/44591/commits/ec35b3f7bc1f777aff363f7b8ad33325e5abc277

File: HLTrigger/Configuration/python/customizeHLTforCMSSW.py [hunk 1]
```diff
@@ -253,6 +253,15 @@ def customizeHLTfor44510(process):
             delattr(producer, 'AXOL1TLModelVersion')
     return process
 
+def customizeHLTfor44222(process):
```
Comment (Martin-Grunewald) [inline HLTrigger/Configuration/python/customizeHLTforCMSSW.py:256]: Why ```44222``` and not ```44591```?

**Review by mmusich — COMMENTED** [review]
> Let me know if it is fine or if I should change something in: https://github.com/cms-sw/cmssw/commit/ec35b3f7bc1f777aff363f7b8ad33325e5abc277

Looks OK to me (I also tested that it works fine with `addOnTests.py -t hlt_mc_GRun` on top of our branch).
To be extremely picky you are customizing for this PR and not for 44222.

File: HLTrigger/Configuration/python/customizeHLTforCMSSW.py [hunk 2]
```diff
@@ -253,6 +253,15 @@ def customizeHLTfor44510(process):
...
+    """
+    Customisation for running HLT with the updated btag info producers from the PR 44222
```
Comment (Martin-Grunewald) [inline HLTrigger/Configuration/python/customizeHLTforCMSSW.py:258]: Same?

File: HLTrigger/Configuration/python/customizeHLTforCMSSW.py [hunk 3]
```diff
@@ -269,6 +278,7 @@ def customizeHLTforCMSSW(process, menuType="GRun"):
     # process = customiseFor12718(process)
 
     process = checkHLTfor43774(process)
+    process = customizeHLTfor44222(process)
```
Comment (Martin-Grunewald) [inline HLTrigger/Configuration/python/customizeHLTforCMSSW.py:281]: Also here?

File: HLTrigger/Configuration/python/customizeHLTforCMSSW.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (stahlleiton) [reply HLTrigger/Configuration/python/customizeHLTforCMSSW.py]: done in a7ab886c1a50d2c052892daf3fd873fd62bfc46a

File: HLTrigger/Configuration/python/customizeHLTforCMSSW.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (stahlleiton) [reply HLTrigger/Configuration/python/customizeHLTforCMSSW.py]: done in a7ab886c1a50d2c052892daf3fd873fd62bfc46a

Comment (stahlleiton) [reply HLTrigger/Configuration/python/customizeHLTforCMSSW.py]: Not sure by "Also here?" but maybe done in a7ab886c1a50d2c052892daf3fd873fd62bfc46a

Comment (mmusich) [reply HLTrigger/Configuration/python/customizeHLTforCMSSW.py]: Can you please squash the last two commits? @stahlleiton . Then we should be good to go.

Comment (stahlleiton) [reply HLTrigger/Configuration/python/customizeHLTforCMSSW.py]: done, the last two commits are squashed in 169cad26493e9982a492a09ae50d3fcfcbc47a8a

Comment (mmusich) [reply HLTrigger/Configuration/python/customizeHLTforCMSSW.py]: thanks a lot!

**Comment by mmusich** [comment]
+hlt
   * PR according description and follow-up review

**Comment by antoniovilela** [comment]
+1

**Pull request merged**